crates/wasm/test/shared.ts

```diff
@@ -1,8 +1,12 @@
-import { initialiseResolver } from "@noir-lang/noir-source-resolver";
+// it's the trick to provide mocha testing. The module `@noir-lang/noir-source-resolver` has no typings to resolve
+// import { initialiseResolver } from "@noir-lang/noir-source-resolver";
+const initialiseResolver =
+    require("@noir-lang/noir-source-resolver").initialiseResolver;
```

Comment:
So the lack of types provided by the `@noir-lang/noir-source-resolver` is the source of the issue? Wouldn't the cleanest solution be to provide types on this?

Reply:
@TomAFrench it should be a good solution for the case. I work with only crate `wasm` and its tests because I was asked to work with this particular crate. 
Now, I think I can switch to this package.

Reply:
The current PR is updated. 
I replaced the declaration of callback in `shared.ts`  taking into account the types declaration in https://github.com/noir-lang/noir-source-resolver/pull/8

[The diff of `shared.ts`](https://github.com/noir-lang/noir/pull/2387/files#diff-9be8753174f6055ae24abc4882a6645672f6dc8e1d92f14bfe4cf3700774eb30L10-R11).